inc/dashboard/namespace.php

```diff
@@ -0,0 +1,456 @@
+ */
+function render_block_column() {
+	global $post;
+	?>
+	<strong><a href="<?php echo esc_url_raw( admin_url( '/admin.php?page=xb-analytics&post=' . $post->ID ) ); ?>"><?php echo esc_attr( $post->post_title ); ?></a></strong>
+	<?php
+}
+
+/**
+ * Render the views column data.
+ */
+function render_views() {
+	global $post;
+	$list = get_views_list( get_days_view() );
+	$views = $list[ $post->post_name ]['views'];
+	?>
+	<div class="post--views"><?php echo number_format_i18n( $views ); ?></div>
+	<?php
+}
+
+/**
+ * Render the average conversion rate column data.
+ */
+function render_average_conversion_rate() {
+	global $post;
+	$list = get_views_list( get_days_view() );
+	$rate = round( $list[ $post->post_name ]['avg_conversion_rate'] * 100 );
+	?>
+	<div class="post--avg-conversion-rate"><?php echo absint( $rate ); ?>%
+	<?php
+}
+
+/**
+ * Return the days url query string.
+ *
+ * @return int The number of days to show insights for. Defaults to 7.
+ */
+function get_days_view() : int {
+	return isset( $_GET['days'] ) ? wp_unslash( absint( $_GET['days'] ) ) : 7;
+}
+
+/**
+ * Calculate average conversion rate.
+ *
+ * @param int $conversions The number of conversions for a block.
+ * @param int $views The number of views for a block.
+ *
+ * @return float The conversion rate (calculated by conversions / views).
+ */
+function calculate_average_conversion_rate( int $conversions = 0, int $views = 0 ) : float {
+
+	// Avoid division by zero.
+	if ( $views === 0 ) {
+		return 0;
+	}
+
+	return $conversions / $views;
+}
+
+/**
+ * Aggregate and map the data from the ES query.
+ *
+ * @param array $buckets The ElasticSearch query data.
+ *
+ * @return array An array of aggregated data.
+ */
+function get_aggregate_data( array $buckets ) : array {
+	$data = [];
+
+	foreach ( $buckets as $block ) {
+		$block_id = $block['key'];
+		$views = $block['views']['uniques']['value'] ?? 0;
+		$conversions = $block['conversions']['uniques']['value'] ?? 0;
+		$conversion_rate = $block['conversionRate']['value'] ?? 0;
+
+		$data[ $block_id ]['views'] = $views;
+		$data[ $block_id ]['conversions'] = $conversions;
+		$data[ $block_id ]['avg_conversion_rate'] = $conversion_rate;
+	}
+
+	return $data;
+}
+
+/**
+ * Sort data by conversion rate.
+ *
+ * @param array $list The array of analytics data by block id.
+ * @param string $order The order to sort by. Accepted values are 'asc' or 'desc'.
+ * @param string $sort The parameter to sort by. Accepted values are 'views' and 'avg_conversion_rate'. Default is 'views'.
+ *
+ * @return array The sorted array.
+ */
+function sort_by( array $list, string $order = 'desc', string $sort = 'views' ) : array {
+	// If an invalid value was passed to $order, default to 'desc'.
+	if ( ! in_array( $order, [ 'asc', 'desc' ], true ) ) {
+		$order = 'desc';
+	}
+
+	// If an invalid value was passed to $sort, default to 'views'.
+	if ( ! in_array( $sort, [ 'avg_conversion_rate', 'views' ], true ) ) {
+		$sort = 'views';
+	}
```

Comment:
I probably wouldn't bother with this validation, it'd error out at the `array_column` bit anyway. Someone using the function might use it incorrectly but not be aware because it sorts things out for them, but potentially makes it behave unexpectedly. I know the available values are in the doc block but I think causing a warning or error is a better thing to do if a function is used wrong.